File: apps/company-knowledge-ui/src/api/feedback.ts
import { apiRequest } from "./client";

export type FeedbackRequest = {
  response_id: string;
  conversation_id?: string | null;
  rating?: "up" | "down" | null;
  flag_gap: boolean;
  comment?: string | null;
  question: string;
  topic_id?: string | null;
};

export type FeedbackResponse = FeedbackRequest & {
  id: string;
  user_id: string;
  tenant_id: string;
  created_at_utc: string;
};

export function submitFeedback(request: FeedbackRequest): Promise<FeedbackResponse> {
  return apiRequest<FeedbackResponse>("/api/v1/feedback", { method: "POST", body: request });
}

File: libs/shared-schemas/python/shared_schemas/feedback.py
from datetime import UTC, datetime
from typing import Literal
from uuid import uuid4

from pydantic import BaseModel, Field


class FeedbackRequest(BaseModel):
    response_id: str = Field(min_length=1, max_length=120)
    conversation_id: str | None = Field(default=None, max_length=120)
    rating: Literal["up", "down"] | None = None
    flag_gap: bool = False
    comment: str | None = Field(default=None, max_length=2000)
    question: str = Field(min_length=1, max_length=1000)
    topic_id: str | None = Field(default=None, max_length=120)


class FeedbackResponse(BaseModel):
    id: str = Field(default_factory=lambda: f"fb-{uuid4().hex[:12]}")
    response_id: str
    conversation_id: str | None = None
    rating: Literal["up", "down"] | None = None
    flag_gap: bool = False
    comment: str | None = None
    question: str
    topic_id: str | None = None
    user_id: str
    tenant_id: str
    created_at_utc: datetime = Field(default_factory=lambda: datetime.now(UTC))

